File: server_py/storage.py
from server_py.db import get_connection, get_dict_cursor


def get_products():
    conn = get_connection()
    try:
        cur = get_dict_cursor(conn)
        cur.execute("SELECT * FROM products")
        return cur.fetchall()
    finally:
        conn.close()


def get_products_by_category(category):
    conn = get_connection()
    try:
        cur = get_dict_cursor(conn)
        cur.execute("SELECT * FROM products WHERE category = %s", (category,))
        return cur.fetchall()
    finally:
        conn.close()


def get_product(product_id):
    conn = get_connection()
    try:
        cur = get_dict_cursor(conn)
        cur.execute("SELECT * FROM products WHERE id = %s", (product_id,))
        return cur.fetchone()
    finally:
        conn.close()


def create_contact_submission(data):
    conn = get_connection()
    try:
        cur = get_dict_cursor(conn)
        cur.execute(
            """INSERT INTO contact_submissions (name, email, phone, company, message)
               VALUES (%s, %s, %s, %s, %s)
               RETURNING *""",
            (data["name"], data["email"], data.get("phone"), data.get("company"), data["message"]),
        )
        result = cur.fetchone()
        conn.commit()
        return result
    finally:
        conn.close()


def get_contact_submissions():
    conn = get_connection()
    try:
        cur = get_dict_cursor(conn)
        cur.execute("SELECT * FROM contact_submissions ORDER BY created_at DESC")
        return cur.fetchall()
    finally:
        conn.close()

File: server/storage.ts
import { db } from "./db";
import { products, contactSubmissions } from "@shared/schema";
import type { Product, InsertProduct, ContactSubmission, InsertContact } from "@shared/schema";
import { eq } from "drizzle-orm";

export interface IStorage {
  getProducts(): Promise<Product[]>;
  getProduct(id: number): Promise<Product | undefined>;
  getProductsByCategory(category: string): Promise<Product[]>;
  createContactSubmission(data: InsertContact): Promise<ContactSubmission>;
  getContactSubmissions(): Promise<ContactSubmission[]>;
}

export class DatabaseStorage implements IStorage {
  async getProducts(): Promise<Product[]> {
    return await db.select().from(products);
  }

  async getProduct(id: number): Promise<Product | undefined> {
    const [product] = await db.select().from(products).where(eq(products.id, id));
    return product;
  }

  async getProductsByCategory(category: string): Promise<Product[]> {
    return await db.select().from(products).where(eq(products.category, category));
  }

  async createContactSubmission(data: InsertContact): Promise<ContactSubmission> {
    const [submission] = await db.insert(contactSubmissions).values(data).returning();
    return submission;
  }

  async getContactSubmissions(): Promise<ContactSubmission[]> {
    return await db.select().from(contactSubmissions);
  }
}

export const storage = new DatabaseStorage();
